app/controllers/users/service_provider_revoke_controller.rb

```diff
@@ -0,0 +1,31 @@
+module Users
+  class ServiceProviderRevokeController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    rescue_from ActiveRecord::RecordNotFound do
+      redirect_to account_url
+    end
+
+    def show
+      @service_provider = ServiceProvider.find(params[:sp_id])
```

Comment:
@solipet you had mentioned having UUIDs for service providers. Do you think it's worth adding a TODO or note for the future here?